### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by tripleee**
Create a simple `Dockerfile` to allow local experimentation with a working basic configuration.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM ruby:2.3
```
Comment (ArtOfCode-) [inline Dockerfile]: We're built against Ruby 2.5 in prod, this should probably match.

File: createdb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+set -e -u
+
+./services
+
+mysql -u root -ppassword <<\____
+
+    # Create the metasmoke database and metasmoke_test database
+
+    CREATE DATABASE metasmoke;
```
Comment (ArtOfCode-) [inline createdb]: This is what `rails db:create` does, you can drop it from the raw SQL.

File: rundb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (ArtOfCode-) [inline rundb]: Why can't we run `foreman start` instead of customising the run script?

Comment (zixuan75) [reply rundb]: My god, we can run this.

File: createdb (hunk 2 above)
Comment (zixuan75) [reply createdb]: Be clarify what are you saying, that does not make sense :-)

**Comment by quartata** [comment]
Poking @tripleee on this, it'd be really useful for testing

**Comment by tripleee** [comment]
Thanks for the ping, I really seem to have way too much in my Github inbox... I'll try to check this tonight.

File: createdb (hunk 2 above)
Comment (tripleee) [reply createdb]: This was just a copy/paste from the instructions in the wiki or the README (don't remember exactly now) but thanks, I'll try to fix it.

File: rundb (hunk 3 above)
Comment (tripleee) [reply rundb]: The `rundb` script does some other things around `foreman start` which Docker does not do for us.  It needs to make sure the MySQL daemon is up, that `rails` listens on the port, etc.

File: createdb (hunk 2 above)
Comment (tripleee) [reply createdb]: There's some sort of dependency here, I can't remember exactly how I arrived at this workaround, but if I remove the SQL, I get permission denied from `db:create`.

```
Starting MariaDB database server: mysqld.
Starting redis-server: redis-server.
/usr/src/app/config/initializers/appconfig.rb:3: warning: already initialized constant AppConfig
/usr/src/app/config/environments/development.rb:21: warning: previous definition of AppConfig was here
Access denied for user 'metasmoke'@'localhost'
Couldn't create database for {"adapter"=>"mysql2", "database"=>"metasmoke", "encoding"=>"utf8", "username"=>"metasmoke", "password"=>"password", "host"=>"127.0.0.1", "port"=>3306}
rails aborted!
Mysql2::Error: Access denied for user 'metasmoke'@'localhost'
/usr/local/bundle/gems/mysql2-0.4.10/lib/mysql2/client.rb:89:in `connect'
/usr/local/bundle/gems/mysql2-0.4.10/lib/mysql2/client.rb:89:in `initialize'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `new'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `mysql2_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:809:in `new_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:853:in `checkout_new_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:832:in `try_to_checkout_new_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:793:in `acquire_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:521:in `checkout'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:380:in `connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1008:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/connection_handling.rb:90:in `connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/mysql_database_tasks.rb:6:in `connection'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/mysql_database_tasks.rb:14:in `create'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:119:in `create'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:139:in `block in create_current'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:316:in `block in each_current_configuration'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:313:in `each'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:313:in `each_current_configuration'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/tasks/database_tasks.rb:138:in `create_current'
/usr/local/bundle/gems/activerecord-5.2.0.rc1/lib/active_record/railties/databases.rake:29:in `block (2 levels) in <top (required)>'
/usr/local/bundle/gems/railties-5.2.0.rc1/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/usr/local/bundle/gems/railties-5.2.0.rc1/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/usr/local/bundle/gems/railties-5.2.0.rc1/lib/rails/command.rb:48:in `invoke'
/usr/local/bundle/gems/railties-5.2.0.rc1/lib/rails/commands.rb:18:in `<top (required)>'
bin/rails:11:in `require'
bin/rails:11:in `<main>'
Tasks: TOP => db:create
(See full trace by running task with --trace)
The command '/bin/sh -c gem install bundler     && bundle install     && ./createdb     && sed -i 's/^web:.*/& -b 0.0.0.0/' Procfile     && yarn install' returned a non-zero code: 1
```

**Comment by tripleee** [comment]
The Dockerfile creates an instance but you then have to "sign up" to create your first account. Then you'll need to connect to the command line to give this account admin privileges.  I would like to make this, too, part of the `Dockerfile` (easier to comment out and run `docker build` again than to discover on your own how to do this) but I don't know yet how to create the first account programmatically.

For the record, the commands after you have signed up are basically

```docker ps -l -q | xargs -i docker exec it {} /bin/sh```

to connect to the latest Docker image you started, and then

```
# rails c
/usr/src/app/config/initializers/appconfig.rb:3: warning: already initialized constant AppConfig
/usr/src/app/config/environments/development.rb:21: warning: previous definition of AppConfig was here
Loading development environment (Rails 5.2.0.rc1)
irb(main):001:0> User.first.add_role(:admin)
   (... lots of output ...)
irb(main):001:0> User.first.add_role(:developer)
   (... lots more output ...)
```

(and then `^D` to exit from the `irb` prompt and `^D` again to exit the shell.)

... But wait, Art is telling me in chat now that there _is_ a way to automate this as well.

**Comment by tripleee** [comment]
This now creates a user `metasmoke@localhost` with password `password` and full admin access.

**Pull request merged**